uast/utils.go

```diff
@@ -17,3 +17,34 @@ func Tokens(n *Node) []string {
 	}
 	return tokens
 }
+
+// CyclomaticComplexity returns the cyclomatic complexity for the node. This uses the method of
+// counting one + one of the following UAST Roles if present on any children:
+// If | SwitchCase |  SwitchDefault | For[Each] | [Do]While | TryCatch | Continue | OpBoolean* | Goto
+// Important: since some languages allow for code defined
+// outside function definitions, this won't check that the Node has the role FunctionDeclarationRole
+// so the user should check that if the intended use is calculating the complexity of a function/method.
+// If the children contain more than one function definitions, the value will not be averaged between
+// the total number of function declarations but given as a total.
+func CyclomaticComplexity(n *Node)  int {
+	complx := 1
+
+	iter := NewOrderPathIter(NewPath(n))
+
+	for {
+		p := iter.Next()
+		if p.IsEmpty() {
+			break
+		}
+		n := p.Node()
+		for _, r := range n.Roles {
+			switch(r) {
+			case If, SwitchCase, SwitchDefault, For, ForEach, While,
```

Comment:
It'd be nice to settle which definition are we going to use, to avoid ambiguities. Are we going to use the original McCabe paper, any other particular implementation?